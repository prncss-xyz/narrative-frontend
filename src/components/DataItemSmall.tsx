import { Dataset } from "../hooks/datasets";
import { Box, Flex, Img } from "./basics";

export function DatasetItemSmall({
  dataset,
  active,
  disabled,
  onClick,
  ...props
}: {
  dataset: Dataset;
  active?: boolean;
  disabled?: boolean;
  onClick?: React.MouseEventHandler;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Flex
      px={3}
      py={1}
      gap={3}
      alignItems="center"
      flexDirection="row"
      borderStyle="solid"
      borderColor={active ? "text" : "bgLight"}
      backgroundColor={active ? "accent" : "bgLight"}
      onClick={disabled ? undefined : onClick}
      cursor={disabled ? undefined : "pointer"}
      {...props}
    >
      <Img height={1} src={dataset.thumbnailUrl} alt="dataset thumbnail" />
      <Flex flexDirection="column">
        <Box color="text" fontSize={2}>
          {dataset.label}
        </Box>
        <Box color="muted" fontSize={0}>
          ${dataset.costPerRecord} per record
        </Box>
      </Flex>
    </Flex>
  );
}
