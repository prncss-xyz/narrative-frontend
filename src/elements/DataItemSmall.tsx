import { Dataset } from "../hooks/datasets";
import { Flex, Img, Box } from "./basics";

export function DatasetItemSmall({
  dataset,
  active,
  disabled,
  onClick,
}: {
  dataset: Dataset;
  active?: boolean;
  disabled?: boolean;
  onClick?: React.MouseEventHandler;
}) {
  return (
    <Flex
      p={1}
      flexDirection="row"
      gap={2}
      border={"solid"}
      borderColor={active ? undefined : "gray3"}
      color={active ? undefined : "gray1"}
      backgroundColor={active ? "white" : "gray3"}
      onClick={disabled ? undefined : onClick}
      css={{
        cursor: disabled ? undefined : "pointer",
      }}
    >
      <Flex minWidth={1} flexDirection="column" justifyContent="center">
        <Img
          src={dataset.thumbnailUrl}
          width={1}
          height={1}
          alt="dataset thumbnail"
        />
      </Flex>
      <Flex flexDirection="column">
        <Box>{dataset.label}</Box>
        <Box color={"gray1"}>${dataset.costPerRecord} per record</Box>
      </Flex>
    </Flex>
  );
}
