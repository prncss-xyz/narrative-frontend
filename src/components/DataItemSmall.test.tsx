import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, expect, it, vi } from "vitest";
import { DatasetItemSmall } from "./DataItemSmall";

const dataset = {
  id: 1,
  name: "fishing",
  label: "Fishing Zones",
  description:
    "Data about global fishing zones, including yields, historical trends and ocean life surveys.",
  thumbnailUrl: "https://picsum.photos/id/1038/92/92",
  costPerRecord: 0.03,
};

describe("DataItemSmall", () => {
  describe("not disabled", async () => {
    const spy = vi.fn();
    const { container } = render(
      <DatasetItemSmall
        dataset={{ ...dataset, label: "Not Disabled" }}
        onClick={spy}
      />
    );
    await userEvent.click(screen.getByText("Not Disabled"));
    it("should trigger when clicked", () => {
      expect(spy).toHaveBeenCalledTimes(1);
    });
    it("should match snapshot", () => {
      expect(container).toMatchSnapshot();
    });
  });

  describe("disabled", async () => {
    const spy = vi.fn();
    const { container } = render(
      <DatasetItemSmall
        dataset={{ ...dataset, label: "Disabled" }}
        onClick={spy}
        disabled
      />
    );
    await userEvent.click(screen.getByText("Disabled"));
    it("should not trigger when clicked", () => {
      expect(spy).toHaveBeenCalledTimes(0);
    });
    it("should match snapshot", () => {
      expect(container).toMatchSnapshot();
    });
  });

  describe("active", () => {
    const { container } = render(
      <DatasetItemSmall dataset={{ ...dataset, label: "Disabled" }} active />
    );
    it("should match snapshot", () => {
      expect(container).toMatchSnapshot();
    });
  });
});
