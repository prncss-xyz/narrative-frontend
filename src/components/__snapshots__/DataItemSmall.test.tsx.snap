// Vitest Snapshot v1

exports[`DataItemSmall > active > should match snapshot 1`] = `<div />`;

exports[`DataItemSmall > disabled > should match snapshot 1`] = `<div />`;

exports[`DataItemSmall > not disabled > should match snapshot 1`] = `<div />`;
